crates/core/src/config/secret.rs

```diff
@@ -0,0 +1,259 @@
+                (path, transport_key)
+            }
+        };
+        let nonce = self.nonce.as_ref().map(read_nonce).transpose()?;
+        let (nonce_path, nonce) = if let Some(nonce) = nonce {
+            (self.nonce.unwrap(), nonce)
+        } else {
+            // try secret dir
+            let path = secrets_dir.join(NONCE_FILENAME);
+            if path.exists() {
+                let nonce = read_nonce(&path)?;
+                (path, nonce)
+            } else {
+                (path, DelegateRequest::DEFAULT_NONCE)
+            }
+        };
+
+        let cipher = self.cipher.as_ref().map(read_cipher).transpose()?;
+
+        let (cipher_path, cipher) = if let Some(cipher) = cipher {
+            (self.cipher.unwrap(), cipher)
+        } else {
+            // try secret dir
+            let path = secrets_dir.join(CIPHER_FILENAME);
+            if path.exists() {
+                let cipher = read_cipher(&path)?;
+                (path, cipher)
+            } else {
+                (path, DelegateRequest::DEFAULT_CIPHER)
+            }
+        };
+
+        Ok(Secrets {
+            transport_keypair,
+            transport_keypair_path,
+            nonce,
+            nonce_path,
+            cipher,
+            cipher_path,
+        })
+    }
+
+    pub(super) fn merge(&mut self, other: Secrets) {
+        self.transport_keypair
+            .get_or_insert(other.transport_keypair_path);
+        self.nonce.get_or_insert(other.nonce_path);
+        self.cipher.get_or_insert(other.cipher_path);
+    }
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct Secrets {
+    #[serde(skip)]
+    pub transport_keypair: TransportKeypair,
+    #[serde(rename = "transport_keypair")]
+    pub transport_keypair_path: PathBuf,
+    #[serde(skip)]
+    pub nonce: [u8; 24],
+    #[serde(rename = "nonce")]
+    pub nonce_path: PathBuf,
+    #[serde(skip)]
+    pub cipher: [u8; 32],
+    #[serde(rename = "nonce")]
+    pub cipher_path: PathBuf,
+}
+
+// Only used in tests
+#[cfg(test)]
+impl Default for Secrets {
+    fn default() -> Self {
+        let transport_keypair = TransportKeypair::new();
+        let nonce = DelegateRequest::DEFAULT_NONCE;
+        let cipher = DelegateRequest::DEFAULT_CIPHER;
+
+        Secrets {
+            transport_keypair,
+            transport_keypair_path: PathBuf::new(),
+            nonce,
+            nonce_path: PathBuf::new(),
+            cipher,
+            cipher_path: PathBuf::new(),
+        }
+    }
+}
+
+impl Secrets {
+    #[inline]
+    pub fn nonce(&self) -> XNonce {
+        *XNonce::from_slice(&self.nonce)
+    }
+
+    #[inline]
+    pub fn cipher(&self) -> XChaCha20Poly1305 {
+        XChaCha20Poly1305::new(GenericArray::from_slice(&self.cipher))
+    }
+
+    #[inline]
+    pub fn transport_keypair(&self) -> &TransportKeypair {
+        &self.transport_keypair
+    }
+
+    pub(super) fn persist(&self) -> std::io::Result<()> {
```

Comment:
we don't need this, we should never be persisting those, either they are passed to the app by argument or we use default ones (or generate o. the flight for the transport key, every new execution can use a different key)